featuretools/primitives/standard/transform/natural_language/num_words.py

```diff
@@ -19,12 +26,18 @@ class NumWords(TransformPrimitive):
 
     name = "num_words"
     input_types = [ColumnSchema(logical_type=NaturalLanguage)]
-    return_type = ColumnSchema(semantic_tags={"numeric"})
+    return_type = ColumnSchema(logical_type=IntegerNullable, semantic_tags={"numeric"})
     compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
     description_template = "the number of words in {}"
 
     def get_function(self):
         def word_counter(array):
-            return array.fillna("").str.count(" ") + 1
+            def _get_number_of_words(elem: pd.Series):
+                """Returns the number of words or null given non-iterable input"""
+                if not isinstance(elem, Iterable):
+                    return pd.NA
+                return sum(1 for word in elem if len(word.strip(punctuation)))
+
+            return array.str.split(DELIMITERS).apply(_get_number_of_words)
```

Comment:
I think we are sort of dividing this primitive logic up between the `word_counter` function and the `_get_number_of_words` function by splitting on `DELIMTERS` before calling `.apply`.

Can we refactor this to call `.apply` on the original series, and add the logic for splitting on `DELIMITERS` to the _get_number_of_words` function instead?

Maybe something like this:
```
    def get_function(self):
        def word_counter(array):
            def _get_number_of_words(elem):
                """Returns the number of words or null given non-iterable input"""
                if not isinstance(elem, str):
                    return pd.NA
                return sum(1 for word in re.split(DELIMITERS, elem) if word.strip(punctuation))

            return array.apply(_get_number_of_words)

        return word_counter
```